### PR TITLE
*: update tikv/client-go

### LINKED ISSUE #43481: storage: backport the stale read enhancement and bug fix to release 6.5
# Enhancement

## The tidb repo

- store/copr: fix missting Request.TxnScope
  - 6.5.3 https://github.com/pingcap/tidb/pull/43553
  - 7.1.0 https://github.com/pingcap/tidb/pull/43374
  - master https://github.com/pingcap/tidb/pull/43367

- planner: recalculate as-of ts of staleread when plan is cached
  - 6.5.3 https://github.com/pingcap/tidb/pull/43550
  - 7.1.0 https://github.com/pingcap/tidb/pull/43257
  - master https://github.com/pingcap/tidb/pull/43204

- grafana: add traffic metrics for staleread traffic
  - 6.5.3 https://github.com/pingcap/tidb/pull/44209
  - 7.1.1 https://github.com/pingcap/tidb/pull/45340
  - master https://github.com/pingcap/tidb/pull/43324

- store/copr: disable stale read when meeting lock for cop
  - 6.5.3 https://github.com/pingcap/tidb/pull/44251
  - 7.1.1 https://github.com/pingcap/tidb/pull/44253
  - master https://github.com/pingcap/tidb/pull/44240

- sessiontxn/staleread: more accurate stale ts
  - 6.5.3 https://github.com/pingcap/tidb/pull/44281
  - 7.1 https://github.com/pingcap/tidb/pull/45760
  - master https://github.com/pingcap/tidb/pull/44272

## The client-go repo

- avoid dataIsNotReady error while retrying stale read on the leader 
  - 6.5.3 https://github.com/tikv/client-go/pull/789
  - 7.1.0 same as master
  - master https://github.com/tikv/client-go/pull/765

- Remove stale-read flag after resolving lock
  - 6.5.3 https://github.com/tikv/client-go/pull/818
  - 7.1 same as master
  - master https://github.com/tikv/client-go/pull/792

- add metrics for stale-read traffic
  - 6.5.3 https://github.com/tikv/client-go/pull/818
  - 7.1.0 same as master
  - master https://github.com/tikv/client-go/pull/776

- batch_client: auto re-connect idle connection when wait connection ready
  - 6.5.3 https://github.com/tikv/client-go/pull/833
  - 7.1 https://github.com/tikv/client-go/pull/665
  - master https://github.com/tikv/client-go/pull/665

- reload region cache when store is resolved from invalid status
  - 6.5.4 https://github.com/tikv/client-go/pull/846
  - 7.1 https://github.com/tikv/client-go/pull/940
  - master https://github.com/tikv/client-go/pull/843

- *: refine non-global stale-read request retry logic
  - 6.5.4 https://github.com/tikv/client-go/pull/851
  - 7.1.1 https://github.com/tikv/client-go/pull/880
  - master https://github.com/tikv/client-go/pull/863

- *: fix stale read ops metric
  - 6.5.4 https://github.com/tikv/client-go/pull/878
  - 7.1.1 https://github.com/tikv/client-go/pull/886
  - master https://github.com/tikv/client-go/pull/889

- handle `mismatch peer id` region error
  - 6.5.4 https://github.com/tikv/client-go/pull/888
  - 7.1.1 https://github.com/tikv/client-go/pull/885 & https://github.com/tikv/client-go/pull/887
  - master https://github.com/tikv/client-go/pull/935

- fix unexpected slow query during GC running after stop 1 tikv-server
  - 6.5.4 https://github.com/tikv/client-go/pull/899
  - 7.1 https://github.com/tikv/client-go/pull/930
  - master https://github.com/tikv/client-go/pull/909

- Resume the max retry check for stale read processing
  - 6.5.4 https://github.com/tikv/client-go/pull/903
  - 7.1 https://github.com/tikv/client-go/pull/940
  - master https://github.com/tikv/client-go/pull/911

- fallback to follower when leader is busy
  - 6.5.4 https://github.com/tikv/client-go/pull/916
  - 7.1 https://github.com/tikv/client-go/pull/940
  - master https://github.com/tikv/client-go/pull/923

- add more log for diagnose
  - 6.5.4 https://github.com/tikv/client-go/pull/914
  - 7.1 https://github.com/tikv/client-go/pull/931
  - master https://github.com/tikv/client-go/pull/915

- add region cache state test & fix some issues of replica selector
  - 6.5.4 https://github.com/tikv/client-go/pull/910
  - 7.1 https://github.com/tikv/client-go/pull/940
  - master https://github.com/tikv/client-go/pull/942

- use context logger as much as possible
  - 6.5 TODO
  - 7.1 https://github.com/tikv/client-go/pull/932
  - master https://github.com/tikv/client-go/pull/908

## The tikv repo

- add gzip compression for check_leader requests
  - 6.5.3 https://github.com/tikv/tikv/pull/14824
  - 7.1 https://github.com/tikv/tikv/pull/14825
  - master https://github.com/tikv/tikv/pull/14721

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

Issue Number: Ref #43481

Problem Summary: update tikv/client-go in go.mod

Related fix:
- https://github.com/tikv/client-go/pull/846
- https://github.com/tikv/client-go/pull/851

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -89,7 +89,7 @@ require (
 	github.com/stretchr/testify v1.8.0
```
Comment (hawkingrei) [inline go.mod:89]: Please run the ```make bazel_prepare```.

**Review by cfzjywxk — APPROVED** [review]

**Review by zyguan — APPROVED** [review]

**Pull request merged**